x-pack/platform/plugins/shared/lens/public/lens_suggestions_api/helpers.ts

```diff
@@ -43,15 +43,14 @@ export function mergeSuggestionWithVisContext({
   }
   const datasourceState = Object.assign({}, visAttributes.state.datasourceStates[datasourceId]);
 
-  // should be based on same columns
+  // the layer columns should exist on the query result
   if (
     !datasourceState?.layers ||
-    Object.values(datasourceState?.layers).some(
-      (layer) =>
-        layer.columns?.some(
-          (c: { fieldName: string }) =>
-            !context?.textBasedColumns?.find((col) => col.id === c.fieldName)
-        ) || layer.columns?.length !== context?.textBasedColumns?.length
```

Comment:
I removed the columns length check, we dont want to revert if the user selected different columns as long as they exist in the response